### PR TITLE
update to 1.6.6 LTS

### PR DESCRIPTION
**Pull request by ngam**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by ngam** [comment]
@conda-forge-admin, please rerender

**Comment by ngam** [comment]
@mkitti, should we target 10.10 then?

```
ld: warning: -pie being ignored. It is only used when linking a main executable
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-fs-poll.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-idna.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-inet.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-loop-watcher.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-random.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-strscpy.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-threadpool.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-timer.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-uv-data-getter-setters.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-uv-common.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-version.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-async.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-core.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-dl.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-fs.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-getaddrinfo.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-getnameinfo.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-loop.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-pipe.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-poll.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-process.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-random-devurandom.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-signal.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-stream.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-tcp.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-thread.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-tty.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-udp.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-bsd-ifaddrs.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-darwin-proctitle.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-darwin.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-fsevents.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-kqueue.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-proctitle.o)) was built for newer macOS version (10.10) than being linked (10.9)
ld: warning: object file (/Users/runner/miniforge3/conda-bld/julia_1648753093737/work/usr/lib/libuv.a(libuv_la-random-getentropy.o)) was built for newer macOS version (10.10) than being linked (10.9)

```

**Comment by isuruf** [comment]
Why not use libuv from conda package?

**Comment by ngam** [comment]
For reference, these are the changes from 1.6.5 to 1.6.6: https://github.com/JuliaLang/julia/compare/v1.6.5...v1.6.6

**Comment by ngam** [comment]
@conda-forge-admin, please rerender

File: .ci_support/linux_64_.yaml [hunk 1]
```diff
@@ -15,17 +15,17 @@ curl:
 cxx_compiler:
 - gxx
 cxx_compiler_version:
-- '9'
+- '10'
```
Comment (ngam) [inline .ci_support/linux_64_.yaml:18]: @isuruf is it okay to downgrade here? I'd like to take this (and c_compiler above) back to 9

I don't see why anything else would've introduced any changes besides these compilers...

**Comment by ngam** [comment]
@conda-forge-admin, please rerender

**Comment by ngam** [comment]
@conda-forge-admin, please rerender

**Comment by ngam** [comment]
@conda-forge-admin, please rerender

**Pull request closed without merging**